### DISCUSSION TITLE
Error: /lib64/libz.so.1: version `ZLIB_1.2.9' not found (required by /vercel/path0/node_modules/canvas/build/Release/libpng16.so.16)

### DISCUSSION THREAD

**Discussion by Mohaiminul007** [vercel/next.js#69034] [Help] [answered]
### Summary

I'm using 'resemblejs' which uses canvas. so deploying in vercel showing errors.
![image](https://github.com/user-attachments/assets/b850732c-1f8f-4971-ad21-2d1e6bb1b127)

Tried all the solutions available here.
- 

### Additional information

```js
tried:

> @napi-rs/canvas this is not an option as i'm not directly using it.
> No deploying form local is also not working.
$vercel build
$vercel deploy
$vercel deploy --prebuilt

> also tried overwriting the build command by:
npm install && npm run build

then also tried:
const nextConfig = {
    reactStrictMode: true,
    experimental: {
        outputFileTracingExcludes: {
            '/node_modules/canvas': true,
        },
    outputFileTracingIgnores: ['canvas'],
    },
};
 
didn't worked.

> using canvas 2.11
> canvas 2.6 is also not working.

so yeah tried a lot of em. 

Looking forward to hear form you guys.
```

**Comment by Wruqe** [comment]
Did you end up finding a solution?

**Reply by Mohaiminul007** [reply]
Just change the Node.js version mine was 18 and I changed it to 20  it worked (sorry for the late reply 😅😅😅 )

**Comment by karliky** [comment]
It does happen to me

**Comment by bunseueng** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/6a45371d-9fb6-4346-8ed4-9ac9bb275ce0)


![photo_2025-01-22_20-13-14](https://github.com/user-attachments/assets/8296f11b-0ee7-4de2-bb5d-d6a3c22c1eef)

I also encountered the same error and fixed it by changing the Node.js version from 22 to 20. I hope this helps you! 😊
